app/file-upload/server/lib/FileUpload.js

```diff
@@ -66,7 +66,7 @@ export const FileUpload = {
 		const room = Rooms.findOneById(file.rid);
 		const directMessageAllowed = settings.get('FileUpload_Enabled_Direct');
 		const fileUploadAllowed = settings.get('FileUpload_Enabled');
-		if (user?.type !== 'app' && canAccessRoom(room, user, file) !== true) {
+		if (user && user.type !== 'app' && canAccessRoom(room, user, file) !== true) {
```

Comment:
@d-gubert I believe this bug was introduced in this commit https://github.com/RocketChat/Rocket.Chat/pull/18995/commits/259af21e0e9015bba658f74835fda1afc38936ba#r516424603